### PR TITLE
Bugfix: onboarding analytics

### PR DESCRIPTION
**Pull request by estebanmino**
**Description**

This PR adds the ability to save analytics events for first time onboarding flow. If the user wants to Start Exploring or Scan QR Code or Import Seedphrase, that event will be saved in state awaiting for the user to Optin or Optout in MetaMetrics. If the user Optin will send the event, if not, it will erase that information.

Related https://github.com/MetaMask/metamask-mobile/issues/930#issuecomment-517320018

**Checklist**

* [ ] There is a related GitHub issue
* [ ] Tests are included if applicable
* [ ] Any added code is fully documented

**Issue**

Resolves #???

### CODE REVIEW

**Comment by bdresser** [comment]
fixes https://github.com/MetaMask/metamask-mobile/issues/978

File: app/components/UI/OptinMetrics/index.js [hunk 1]
```diff
@@ -180,19 +199,26 @@ class OptinMetrics extends PureComponent {
...
+		});
 		this.continue();
 	};
 
 	/**
 	 * Callback on press confirm
 	 */
 	onConfirm = async () => {
-		await AsyncStorage.setItem('@MetaMask:metricsOptIn', 'agreed');
+		//	await AsyncStorage.setItem('@MetaMask:metricsOptIn', 'agreed');
```
Comment (brunobar79) [inline app/components/UI/OptinMetrics/index.js]: ?

File: app/components/Views/ImportWallet/index.js [hunk 2]
```diff
@@ -324,24 +353,17 @@ class ImportWallet extends PureComponent {
 			);
 			return false;
 		}
-
-		if (this.props.navigation.getParam('existingUser', false)) {
```
Comment (brunobar79) [inline app/components/Views/ImportWallet/index.js]: Why is this deleted?

File: app/reducers/onboarding/index.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
+import { REHYDRATE } from 'redux-persist';
+
+const initialState = {
+	event: undefined
+};
+
+/**
+ * Reducer to keep track of user oboarding actions to send it to analytics if thw user
```
Comment (brunobar79) [inline app/reducers/onboarding/index.js]: typo "if THE user"

File: app/reducers/onboarding/index.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+	switch (action.type) {
+		case REHYDRATE:
+			if (action.payload && action.payload.onboarding) {
+				return { ...state, ...action.payload.onboarding };
+			}
+			return state;
+		case 'SAVE_EVENT':
+			return {
+				...state,
+				event: action.event
```
Comment (brunobar79) [inline app/reducers/onboarding/index.js]: I'd prefer if this was an array instead and you push events to the stack, so you can track the scenario of the user going to import wallet and then coming back to the first screen and then create a new wallet.

File: app/components/Views/ImportWallet/index.js (hunk 2 above)
Comment (estebanmino) [reply app/components/Views/ImportWallet/index.js]: because we were doing the same thing here https://github.com/MetaMask/metamask-mobile/pull/976/files#diff-e78d44598405f73efa09c3b995881f68R340 before calling this `onPressSync` method

Comment (estebanmino) [reply app/components/Views/ImportWallet/index.js]: in fact this line is using a param that we don't provide when doing a navigation.push so this will always be false

File: app/components/UI/OptinMetrics/index.js (hunk 1 above)
Comment (estebanmino) [reply app/components/UI/OptinMetrics/index.js]: my bad, debug code don't modified good call

File: app/reducers/onboarding/index.js (hunk 4 above)
Comment (estebanmino) [reply app/reducers/onboarding/index.js]: totally agree, added a comment on this on analytics sheet. As we didn't have this array analytics behavior there yet I chose to wait for https://github.com/MetaMask/Design/issues/119

File: app/components/UI/OptinMetrics/index.js [hunk 5]
```diff
@@ -192,7 +214,11 @@ class OptinMetrics extends PureComponent {
 	onConfirm = async () => {
 		await AsyncStorage.setItem('@MetaMask:metricsOptIn', 'agreed');
 		Analytics.enable();
-		Analytics.trackEvent(ANALYTICS_EVENT_OPTS.ONBOARDING_METRICS_OPT_IN);
+		InteractionManager.runAfterInteractions(() => {
+			this.props.event && Analytics.trackEvent(this.props.event);
```
Comment (brunobar79) [inline app/components/UI/OptinMetrics/index.js]: Assuming you have an array of events instead:

```
if(this.props.events && this.props.events.length){
    this.props.events.forEach(e => Analytics.trackEvent(e));
}
```

**Review by brunobar79 — APPROVED** [review]
💯

**Comment by ibrahimtaveras00** [comment]
The events that are saved and triggered later are not being shown in mixpanel

As seen here (in this example I did import via seed phrase, but also happens with start exploring and sync) 

image

This example is from an android device, but same thing happens on iOS as well

**Comment by ibrahimtaveras00** [comment]
Looks good 👍
